content/hardware/05.pro-solutions/solutions-and-kits/portenta-machine-control/tutorials/plc-ide-setup-license/content.md

```diff
@@ -100,21 +100,54 @@ Click OK to save the settings and press the connect button on **On-line > connec
 
 ![Connect to the board](assets/connect.png)
 
-### 6. License Activation
+### 6. License Activation With Product Key (Portenta Machine Control)
 
 If the communication is successful the main panel will show the license section to be filled in and a status indicator.
 
-To use your product license, fill it in on the form and press the **Activate** button.
+#### Online Activation
+
+To use your product license, paste the key in the blank next to **Product Key:** and press the **Activate** button.
 
 ![Connection success](assets/licenseActivation.png)
 
-After you click the **Activate** button, it will use the license and attach it to the board. A popup telling you to reboot the target (which is the device connected to the PLC IDE) will show up once the attachment finishes. Press the reset button of your device to perform the reboot.
+After you click the **Activate** button, the IDE will bind the license to the hardware ID. A popup telling you to reboot the target (which is the device connected to the PLC IDE) will show up: press the reset button of your device to reboot.
 
 ![Activation succeed](assets/connectedAndActivated.png)
 
-Once the board has been activated, it is recommended to get the **Hardware-ID** of your device. 
+Once the product has been activated, it is recommended to get the **Hardware-ID** of your device. 
+
+#### Offline Activation
+
+To activate the product Offline, deselect the "online" slider and the scenario below will show up:
+
+![Activation Offline](https://user-images.githubusercontent.com/100422525/225364908-b2d0e497-014d-4fe4-bef0-34c846cb5103.png)
+
+Follow the instructions provided, using our [Contact form](https://www.arduino.cc/en/contact-us) and generating the **Activation Request File**.
```

Comment:
That comma as all the commas is extremely subjective but I would say the phrase is short enough to get rid of it.

```suggestion
Follow the instructions provided using our [Contact form](https://www.arduino.cc/en/contact-us) and generating the **Activation Request File**.
```